plugins/inputs/directory_monitor/directory_monitor.go

```diff
@@ -0,0 +1,422 @@
+package directory_monitor
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/plugins/parsers/csv"
+	"github.com/influxdata/telegraf/selfstat"
+	"golang.org/x/sync/semaphore"
+	"gopkg.in/djherbis/times.v1"
+
+	"compress/gzip"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const sampleConfig = `
+  ## The directory to monitor and read files from.
+  directory = ""
+  #
+  ## The directory to move finished files to.
+  finished_directory = ""
+  #
+  ## The directory to move files to upon file error.
+  ## If not provided, erroring files will stay in the monitored directory.
+  # error_directory = ""
+  #
+  ## The amount of time a file is allowed to sit in the directory before it is picked up.
+  ## This time can generally be low but if you choose to have a very large file written to the directory and it's potentially slow,
+  ## set this higher so that the plugin will wait until the file is fully copied to the directory.
+  # directory_duration_threshold = "50ms" 
+  #
+  ## A list of the only file names to monitor, if necessary. Supports regex. If left blank, all files are ingested.
+  # files_to_monitor = [".*.csv"]
```

Comment:
this is a misleading regex example. it will match the file "foo.notcsv"

Reply:
ha, yup